Widgets/Animation/VarmintWidgetAnimationCollection.cs

```diff
@@ -57,15 +57,11 @@ public partial class VarmintWidgetAnimation
         {
             return new VarmintWidgetAnimation(durationSeconds, (widget, delta) =>
             {
-
-                // Animations scales from 1 (current size) to scale factor
-                float changeFactor = 0;
                 if (scaleFactor < 0) throw new ArgumentException();
-                if (scaleFactor < 1) changeFactor = -scaleFactor;
-                if (scaleFactor >= 1) changeFactor = 1 - scaleFactor;
-
-                widget.Size = originalSize * (1 + (changeFactor * (float)delta));
 
+                // Scales from 1 (current size) to scale factor
+                var sizeChange = originalSize * scaleFactor - originalSize; 
```

Comment:
You can optimize this code by moving this line to just before the return.  That way it is only calculated once instead of every frame.